backend/app.js

```diff
@@ -134,46 +136,144 @@ async function getCabinet(cabinetIdx) {
+  }
 }
 
 // lent 테이블에서 사물함 정보 삭제
-async function deleteLent(connection, userLentInfo) {
-  await connection.query(
-    `
+async function deleteLent(userLentInfo) {
+  let connection;
+  try {
+    connection = await pool.getConnection();
+    await connection.query(
+      `
     DELETE 
     FROM lent 
     WHERE lent_cabinet_id=${userLentInfo.lent_cabinet_id}
     `
-  );
+    );
+  } catch (err) {
+    console.log(err);
+    throw err;
+  } finally {
+    connection.release();
+  }
 }
 
 // lent_log에 반납되는 사물함 정보 추가
-async function addLentLog(connection, userLentInfo) {
-  await connection.query(
-    `
-    INSERT INTO lent_log(log_cabinet_id, log_user_id, lent_time, return_time) 
-    VALUES (${userLentInfo.lent_cabinet_id}, ${userLentInfo.lent_user_id}, '${userLentInfo.lent_time}', now())
-    `
-  );
+async function addLentLog(userLentInfo) {
+  let connection;
+  try {
+    connection = await pool.getConnection();
+    await connection.query(
+      `
+      INSERT INTO lent_log(log_cabinet_id, log_user_id, lent_time, return_time) 
+      VALUES (${userLentInfo.lent_cabinet_id}, ${userLentInfo.lent_user_id}, '${userLentInfo.lent_time}', now())
+      `
+    );
+  } catch (err) {
+    console.log(err);
+    throw err;
+  } finally {
+    connection.release();
+  }
 }
 
 // 사물함 activation 상태 변경
-async function modifyCabinetActivation(connection, cabinetIdx, activation) {
-  const content = `
-  UPDATE cabinet c
-  SET activation=${activation}
-  WHERE cabinet_id=${cabinetIdx}
-  `;
-  await connection.query(content);
+async function modifyCabinetActivation(cabinetIdx, activation) {
+  let connection;
+  try {
+    connection = await pool.getConnection();
+    const content = `
+    UPDATE cabinet c
+    SET activation=${activation}
+    WHERE cabinet_id=${cabinetIdx}
+    `;
+    await connection.query(content);
+  } catch (err) {
+    console.log(err);
+    throw err;
+  } finally {
+    connection.release();
+  }
+}
+
+async function searchIntraId(intraId) {
+  let connection;
+  try {
+    connection = await pool.getConnection();
+    const content = `
+    SELECT u.intra_id, c.location, c.section, c.floor, c.cabinet_num, l.lent_time, l.expire_time
+    FROM user u
+    JOIN lent l
+    ON u.user_id=l.lent_user_id
+    JOIN cabinet c
+    ON l.lent_cabinet_id=c.cabinet_id
+    WHERE u.intra_id='${intraId}'
+    union
+    SELECT u.intra_id, c.location, c.section, c.floor, c.cabinet_num, ll.lent_time, ll.return_time as expire_time
+    FROM user u
+    JOIN lent_log ll
+    ON u.user_id=ll.log_user_id
+    JOIN cabinet c
+    ON ll.log_cabinet_id=c.cabinet_id
+    WHERE u.intra_id='${intraId}'
+    ORDER BY lent_time DESC;
+    `;
+    const result = await connection.query(content);
+    console.log("=====searchIntraId=====");
+    console.log(result);
+    return result;
+  } catch (err) {
+    console.log(err);
+    throw err;
+  } finally {
+    connection.release();
+  }
+}
+
+async function searchCabinetNum(cabinetNum, floor) {
```

Comment:
```suggestion
async function getInfoByCabinetNum(cabinetNum, floor) {

```

getIfnoByCabinetNum이 좀 더 명확한거 같습니다!